Feature/skip button

agarciabz opened the pull request:
Adding small skip button, pending to restyle.

agarciabz committed:
Add connection to endpoint

agarciabz committed:
Add package-lock.json

agarciabz committed:
Merge branch 'develop' of https://github.com/rogue-sound/rogue-sound-web into feature/skip-button

agarciabz committed:
Add skip button as an icon

agarciabz committed:
Fixed eslint errors

Commit:
Move function inside component

Commit:
Update azure functions url

Commit:
Merge branch 'develop' of https://github.com/rogue-sound/rogue-sound-web into feature/skip-button

bonavida commented on src/Layout/Footer/Footer.js:
```diff
@@ -1,15 +1,27 @@
 const Footer = () => {
   const current = useSelector(state => state.playing.current);
 
+  const skipHandler = async () => {
```
You don't need to do this right now, but it would be interesting to use the `useCallback` hook here to prevent unnecessary renders.

The main idea of the `useCallback` hook is that it will return a memoized version of the callback that only changes if one of the dependencies has changed.

agarciabz replied:
Done.

bonavida commented on src/Layout/Footer/Footer.js:
```diff
@@ -18,6 +30,12 @@ const Footer = () => {
+        <FontAwesomeIcon
+          icon="forward"
+          title="Skip current song"
+          className="skip-button"
```
I know we're currently using CSS classes to style Font Awesome icons, but maybe it would be great to start using `styled-components` for that.

Something like this:
```js
import React from 'react';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const StyledIcon = styled(FontAwesomeIcon).`
  color: red;
`;

const Icon = () => (
   <StyledIcon
      icon={['fas', 'balance-scale']}
      fixedWidth
      size="5x"
   />
 );

export default Icon;
```

cesarandex replied:
I'd keep it like this for the PR and address it in a standalone PR which would update all current icons if we end up going for it

agarciabz replied:
Leaving for the next PR then.

bonavida commented on src/config/config.js:
I'm not quite sure if this should be changed.

cc/ @jmolla31

jmolla31 replied:
It's ok fttb, but we should move that into an environment variable or the .env file.

cesarandex replied:
Agree, for now I'd keep our usual URL and make a PR to be selected through the .env file

agarciabz replied:
Leaving for the next PR then.

bonavida reviewed (commented):
The code looks really good, but I've left a few comments that I think you should address. Please, give it a look and I'll review the PR again!

cesarandex commented on src/Layout/Footer/Footer.js:
```diff
@@ -1,15 +1,27 @@
 
+  const skipHandler = async () => {
+    if (current && current.publicId) {
+      const roomSessionModel = { roomSessionId: current.publicId };
```
Something seems off here for me although I'm not sure on what the endpoint expects to receive. We're sending an object with a roomSessionId property but assigning the current song ID

@pabravil can you provide more information?

agarciabz replied:
You're right, should provide the current session ID, yet it's still working.

agarciabz replied:
For the moment I'll leave it without parameters and we'll see later how to handle session ID.

cesarandex reviewed (commented):
I like it, just look at the comments so we can make some small improvements and it's good to go! (Although eventually we'll have to add admin logic for the button or something)

agarciabz committed:
Update function with useCallback hook

agarciabz committed:
Removing parameter roomSession

Leaving the post method without arguments because the endpoint doesn't
need them for the moment.

agarciabz commented:
Can you do something about the prod build? It's working fine on my machine but here I'm not sure why it's not working.
See trace:

ERROR in ./src/assets/svg/devices.svg (./node_modules/@svgr/webpack/lib?-svgo,+titleProp,+ref!./src/assets/svg/devices.svg)
Module build failed (from ./node_modules/@svgr/webpack/lib/index.js):
SyntaxError: Invalid regular expression: /[_.\- ]+([\p{Alpha}\p{N}_]|$)/: Invalid escape
    at camelCase (/home/runner/work/rogue-sound-web/rogue-sound-web/node_modules/camelcase/index.js:69:3)
    at getComponentName (/home/runner/work/rogue-sound-web/rogue-sound-web/node_modules/@svgr/core/lib/state.js:18:53)
    at expandState (/home/runner/work/rogue-sound-web/rogue-sound-web/node_modules/@svgr/core/lib/state.js:26:43)
    at run (/home/runner/work/rogue-sound-web/rogue-sound-web/node_modules/@svgr/core/lib/convert.js:13:48)
    at convert (/home/runner/work/rogue-sound-web/rogue-sound-web/node_modules/@svgr/core/lib/convert.js:26:10)
    at <anonymous>
 @ ./src/Layout/Header/DeviceSelector/DeviceSelector.js 32:0-108 100:38-49
 @ ./src/Layout/Header/DeviceSelector/index.js
 @ ./src/Layout/Header/Header.js
 @ ./src/Layout/Header/index.js
 @ ./src/Layout/Layout.js
 @ ./src/Layout/index.js
 @ ./src/routes/routes.js
 @ ./src/app.js
 @ ./src/index.js
 @ multi babel-polyfill react-hot-loader/patch ./src/index.js

bonavida commented:
@agarciabz I think it has something to do with the Node version we are using for the prod build: https://github.com/rogue-sound/rogue-sound-web/blob/873506afe1c91ae0129867aa6fe5d5dfb7645d5d/.github/workflows/nodejs.yml#L32

@jmolla31 Why are we using node 8? We should be using LTS version.

jmolla31 committed:
Bump node version in github workflow

jmolla31 commented:
@bonavida Bumped the node version to 12.x and it seems to be working again... 🚀

agarciabz committed:
Merge branch 'develop' into feature/skip-button

agarciabz committed:
Switch file names

agarciabz committed:
Dependencies with npm 12
